invenio_collections/models.py

```diff
@@ -360,6 +367,14 @@ def sort_methods(self):
 
         return BsrMETHOD.query.order_by(BsrMETHOD.name).all()
 
+    def can_be_viewed_by(self, user_info):
+        """Check if the user has sufficient rights to view the collection."""
+        if not self.is_restricted():
+            return True
+        (auth_code, auth_msg) = acc_authorize_action(
```

Comment:
Please note that caching should be implemented as calling `acc_authorize_action` is pretty "expensive". Have a look at `precached_permitted_restricted_collections` key in `user_info` first.